owners/src/owners.js

```diff
@@ -92,11 +92,22 @@ class OwnersParser {
    * @return {OwnersRule} parsed OWNERS file rule.
    */
   async parseOwnersFile(ownersPath) {
-    const contents = this.localRepo.readFile(ownersPath);
+    const contents = await this.localRepo.readFile(ownersPath);
     const ownersList = yaml.parse(contents);
 
     return new OwnersRule(ownersPath, ownersList);
   }
+
+  /**
+   * Parse all OWNERS rules in the repo.
+   *
+   * @return {OwnersRule[]} a list of all rules defined in the local repo.
+   */
+  async parseAllOwners() {
+    const ownersPaths = await this.localRepo.findOwnersFiles();
+    return await Promise.all(ownersPaths.map(
+        ownersPath => this.parseOwnersFile(ownersPath)));
```

Comment:
Would this cause the OWNERS.yaml files to be read for every file modified in the PR?

Reply:
This would build an owners tree with all owners files, yes. We could read fewer files, and maybe later I'll come back in to filter this down, but reading them once per PR shouldn't be too expensive. I'll leave a TODO here to restrict the searching to paths modified in the PR, and can touch that once I start working with the actual PR files.